crawl_OAG.py

```diff
@@ -97,49 +107,188 @@ def crawl_helper(professor, university, file):
+
+    # Check for empty file path
+    if file == "":
+        return authors
+
+    # Open file
+    file_papers = open(file, 'r')
+
+    # Loop through file.
+    while True:
+        # Get next line from file
+        line = file_papers.readline()
+    
+        # If line is empty end of file is reached
+        if not line:
+            break
+
+        # Load the line into json
+        author_json = json.loads(line)
+        temp_dict = {}   
+
+        if "id" in author_json:
+            temp_dict["id"] = author_json["id"]
+
+        else:
+            temp_dict["id"] = ""
+
+        if "name" in author_json:
+            temp_dict["name"] = author_json["name"]
+
+        else:
+            temp_dict["name"] = ""
+
+        if "orgs" in author_json:
+            temp_dict["org"] = author_json["orgs"]
+
+        else:
+            temp_dict["org"] = ""
+
+        if "pubs" in author_json:
+            temp_dict["pubs"] = author_json["pubs"]
+
+        else:
+            temp_dict["pubs"] = ""
+            
+        authors = authors.append(temp_dict, ignore_index=True)
+    
+    return authors
+
+def open_ssh_tunnel(verbose=False):
+    """Open an SSH tunnel and connect using a username and password.
+    
+    Args:
+      verbose(bool): Set to True to show logging
+
+    Returns: 
+      tunnel: Global SSH tunnel connection
+    """
+    # SSH Information
+    ssh_host = 'Owl2.cs.illinois.edu'
+    ssh_user = 'bm12'
+
+    # Load SSH password from file
+    text_file = open("ssh_password.txt", "r")
+    ssh_pass = text_file.read()
+    text_file.close()
+
+    if verbose:
+        sshtunnel.DEFAULT_LOGLEVEL = logging.DEBUG
+    
+    global tunnel
+    tunnel = SSHTunnelForwarder(
+        (ssh_host, 22),
+        ssh_username = ssh_user,
+        ssh_password = ssh_pass,
+        remote_bind_address = ('127.0.0.1', 3306)
+    )
+    
+    tunnel.start()
+
+def mysql_connect():
+    """Connect to a MySQL server using the SSH tunnel connection
+    
+    Returns: 
+      connection: Global MySQL database connection
+    """
+    # Database Credentials
+    db_host = '127.0.0.1'
+    db_name = "bm12_publications"
+    db_user = "bm12"
+
+    # Load database password from file
+    text_file = open("db_password.txt", "r")
+    db_password = text_file.read()
+    text_file.close()
+
+    global connection
+    
+    connection = pymysql.connect(
+        host=db_host,
+        user=db_user,
+        passwd=db_password,
+        db=db_name,
+        port=tunnel.local_bind_port
+    )
+
+def run_query(sql):
+    """Runs a given SQL query via the global database connection.
+    
+    Args:
+      sql: MySQL query
+    
+    Returns: 
+      Pandas dataframe containing results
+    """
+    
+    return pd.read_sql_query(sql, connection)
+
+def mysql_disconnect():
+    """Closes the MySQL database connection.
+    """
+    connection.close()
+
+def close_ssh_tunnel():
+    """Closes the SSH tunnel connection.
+    """
+    tunnel.close
```

Comment:
Same thing here, did you mean `tunnel.close()`?

Reply:
```suggestion
    tunnel.close()
```

Reply:
Updated to be tunnel.close()